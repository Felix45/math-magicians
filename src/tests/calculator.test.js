import calculate from '../logic/calculator';

describe('Tests on calculator.js calculate function', () => {
  test('Test AC button on calculator', () => {
    const result = calculate({ total: '777', next: '75', operation: '' }, 'AC');
    expect(result).toStrictEqual({ total: null, next: null, operation: null });
  });

  test('Test x button on calculator', () => {
    const result = calculate({ total: '1000', next: '75', operation: 'x' }, '=');
    expect(result).toStrictEqual({ total: '75000', next: null, operation: null });
  });

  test('Test +/- button on calculator', () => {
    const result = calculate({ total: -12, next: null, operation: null }, '+/-');
    expect(result).toStrictEqual({ total: '12', next: null, operation: null });
  });

  test('Test dot button on calculator', () => {
    let result = calculate({ total: null, next: null, operation: null }, '5');
    result = calculate(result, '.');
    result = calculate(result, '100');
    expect(result).toStrictEqual({ total: null, next: '5.100' });
  });

  test('Test - button on calculator', () => {
    const result = calculate({ total: '1000', next: '1200', operation: '-' }, 'x');
    expect(result).toStrictEqual({ total: '-200', next: null, operation: 'x' });
  });

  test('Test + button on calculator', () => {
    const result = calculate({ total: '1000', next: '1200', operation: '-' }, '+');
    expect(result).toStrictEqual({ total: '-200', next: null, operation: '+' });
  });

  test('Test รท button on calculator', () => {
    const result = calculate({ total: '1200', next: '1200', operation: 'รท' }, '=');
    expect(result).toStrictEqual({ total: '1', next: null, operation: null });
  });

  test('Test % button on calculator', () => {
    const result = calculate({ total: null, next: '1200', operation: null }, '%');
    expect(result).toStrictEqual({ total: '1200', next: null, operation: '%' });
  }); 
});
