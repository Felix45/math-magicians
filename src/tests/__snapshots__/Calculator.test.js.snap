// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="container d-flex space-between"
>
  <h1
    className="title"
  >
    Let's do some math!
  </h1>
  <div
    className="d-flex flex-column calculator"
  >
    <div
      className="bg-dark d-flex flex-end p-1"
    >
      0
    </div>
    <div
      className="d-flex calculator-btns"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</section>
`;
